src/frontend/src/components/FillForms/UserDataForm.tsx

```diff
@@ -41,297 +43,273 @@ class UserDataForm extends React.Component<Props, State> {
     } else {
       return "";
     }
-  };
+  }
 
-  private buildDob = () => {
-    if (this.props.aliases.length > 0) {
-      const firstAlias = this.props.aliases[0];
+  function buildDob() {
+    if (aliases.length > 0) {
+      const firstAlias = aliases[0];
       return firstAlias.birth_date;
     } else {
       return "";
     }
-  };
-
-  public state: State = {
-    name: this.buildName(),
-    dob: this.buildDob(),
-    mailingAddress: "",
-    phoneNumber: "",
-    city: "",
-    state: "",
-    zipCode: "",
-    invalidZipCode: false,
-    invalidPhone: false,
-    invalidBirthDate: false,
-  };
-
-  phoneNumberPattern = new RegExp(".*[0-9].*");
-  zipCodePattern = new RegExp("[0-9][0-9][0-9][0-9][0-9].*");
-
-  public validateForm = () => {
-    const phoneNumberMatch = this.phoneNumberPattern.exec(
-      this.state.phoneNumber
-    );
-    const zipCodeMatch = this.zipCodePattern.exec(this.state.zipCode);
-    return new Promise((resolve: (value?: unknown) => void) => {
-      this.setState<any>(
-        {
-          invalidZipCode: this.state.zipCode.length > 0 && !zipCodeMatch,
-          invalidPhone: this.state.phoneNumber.length > 0 && !phoneNumberMatch,
-          invalidBirthDate:
-            this.state.dob.length > 0 &&
-            moment(this.state.dob, "M/D/YYYY", true).isValid() === false,
-        },
+  }
 
-        resolve
-      );
-    });
-  };
+  function emptyFieldsCheck() {
+    const emptyZipCode = zipCode.length === 0;
+    const emptyPhoneNumber = phoneNumber.length === 0;
+    const emptyDob = dob.length === 0;
+    if (emptyZipCode && emptyPhoneNumber && emptyDob) {
+      setModalClose(false);
+    } else {
+      return false;
+    }
+  }
 
-  public handleChange = (e: React.BaseSyntheticEvent) => {
-    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
-    // using the "any" type.
-    this.setState<any>({
-      [e.target.id]: e.target.value,
-    });
-  };
+  function validateForm() {
+    const phoneNumberPattern = new RegExp(".*[0-9].*");
+    const zipCodePattern = new RegExp("[0-9][0-9][0-9][0-9][0-9].*");
```

Comment:
Could we add a comment with an example that matches and an example that barely doesn't?